pkg/orchestration/wiring/postgres/plugin.go

```diff
@@ -164,13 +159,26 @@ func instanceSpec(resource *orch_v1.StateResource, context *wiringplugin.WiringC
 
 	// if this postgres depends on Dedicated RDS, it means it should be created there instead on the Default RDS
 	// there should be only one RDS dependency
-	rdsDependency, err := getRDSDependency(context.Dependencies)
+	dep, found, err := context.FindTheOnlyDependency()
+	if err != nil {
+		return nil, errors.Wrap(err, "postgres resources can only depend on one RDS resource type")
+	}
+
+	// Did not find any dependencies
+	if !found {
+		return json.Marshal(finalSpec)
+	}
+
+	// Check if dependency has a RDS shape
+	_, found, err = knownshapes.FindRDSShapes(dep.Contract.Shapes)
 	if err != nil {
-		return nil, err
+		return json.Marshal(finalSpec)
```

Comment:
This should return an error if there IS a dependency but something is wrong with its shape

Reply:
```suggestion
		return nil, errors.Wrapf(err, "unable to determine if shape %s db was a dependency", knownshapes.SharedDbShape)
```

Reply:
Does the suggestion above meet your expectations?